src/Api/ErrorParser/JsonParserTrait.php

```diff
@@ -15,12 +15,26 @@ trait JsonParserTrait
     private function genericHandler(ResponseInterface $response)
     {
         $code = (string) $response->getStatusCode();
+        if ($this->api &&
+            $this->api->getMetadata('awsQueryCompatible') &&
+            $response->getHeaderLine('x-amzn-query-error')
+        ) {
+            $queryError = $response->getHeaderLine('x-amzn-query-error');
+            $parts = explode(';', $queryError);
+            if (isset($parts) && count($parts) == 2 && $parts[0] && $parts[1]) {
```

Comment:
Since this is on explode, if there are two parts, won't the array always be indexed as normal and the $parts[0] and $parts[1] check is not necessary.

Reply:
The `parts[0]` and `parts[1]` checks are to make sure they are not empty strings

Reply:
Ah, that won't catch whitespace only, so maybe add trim to it if you want to catch that as well

Reply:
copy. will add trim()! thanks

Reply:
actually getHeaderLine should trim this, so I think it's good

Reply:
Cool. I've already submitted the commit containing the `trim()`. while it doesn't hurt to trim again, let me know if you wanna me to revert that commit or leave as it is.